### PR TITLE
refactor: refactor and fix issues with tooltips

### PR DESCRIPTION
**Pull request by anxolin**
# Summary

Closes https://github.com/cowprotocol/cowswap/issues/4343 

This PR unifies the tooltips of CoW Swap.

## The problem

There was multiple implementations of tooltip, with a lot of repetition. This made https://github.com/cowprotocol/cowswap/pull/4339 (and the previous PRs on that series) to not get the UX improvements for all tooltips (this was showing some inconsistent behaviour)

This PR simplifies the use of the tooltips and makes sure all the usage is coherent. 

## Summary of tooltips
The main tooltips to use now are these, please consider using them in this decreasing order: 
(all in `@cowprotocol/ui`)
- `HelpTooltip`: Tooltip to use for help icons with a tooltip that shows on hover
- `InfoTooltip`: Tooltip to show a info icon with a tooltip that shows on hover
- `HoverTooltip`: Tooltip that appears when hovering over the children you pass
   - Sometimes you might want to add the `wrapInContainer` (mostly when you add text and want to limit the width)
   - Useful when you want to control the content instead of showing an icon
- `Tooltip`: Tooltip that displays text when the passed `show` prop is true.
   - Most of the times, you won't need this one
   -  Useful if you want to control the show state

## Test
This PR affects to all the app, so we should test carefully
- Check all tooltips
- Everything should work as in production or better (with the improvements of https://github.com/cowprotocol/cowswap/pull/4339 )

### CODE REVIEW

**Review by elena-zh — APPROVED** [review]
Hey @anxolin , thank you! Everything works great in the desktop version. 

However, still there are some issues related to tooltips displaying in mobile devices. I've opened #4348 issue for this (in order not to block the release).

**Comment by anxolin** [comment]
@elena-zh I just decided to address also the scroll thing you mentioned. For me it was okeish we don't hide them and hidden them means we add a bit more of complexity, but since its not too much I added it. 

Let me know if you prefer this new version

**Comment by elena-zh** [comment]
@anxolin , works for me, thank you!

File: apps/cowswap-frontend/src/modules/tokensList/pure/FavouriteTokensList/index.tsx [hunk 1]
```diff
@@ -18,8 +16,8 @@ export function FavouriteTokensList(props: FavouriteTokensListProps) {
   return (
     <div>
       <styledEl.Header>
-        <h4>Favorite tokens</h4>
-        <InfoIcon iconType="help" content="Your favorite saved tokens. Edit this list in your account page." />
+        <h4>Favorite tokens</h4>
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/tokensList/pure/FavouriteTokensList/index.tsx:19]: Nitpick: extra space 🤓 

```suggestion
        <h4>Favorite tokens</h4>
```

File: apps/cowswap-frontend/src/modules/tokensList/pure/TokenTags/index.tsx [hunk 2]
```diff
@@ -70,12 +70,12 @@ function TagDescriptor({ tags, children }: { children?: React.ReactNode; tags: T
   return (
     <styledEl.TagContainer>
       {tags.map((tag) => (
-        <MouseoverTooltip key={tag.id} text={tag.description}>
+        <HoverTooltip wrapInContainer  key={tag.id} content={tag.description}>
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/tokensList/pure/TokenTags/index.tsx:73]: Nitpick: another extra space

```suggestion
        <HoverTooltip wrapInContainer key={tag.id} content={tag.description}>
```


Sorry, I cannot not see this 🙈

File: apps/cowswap-frontend/src/modules/twap/containers/TwapFormWarnings/warnings/SwapPriceDifferenceWarning.tsx [hunk 3]
```diff
@@ -49,7 +49,7 @@ export function SwapPriceDifferenceWarning({
     <InlineBanner bannerType={'savings'}>
       {isTwapBetter ? (
         <>
-          <strong>Maximizing Your Gains! {/*<InfoIcon content="TODO: set the tooltip content"></InfoIcon>*/}</strong>
+        <strong>Maximizing Your Gains! </strong>
```
Comment (alfetopito) [inline apps/cowswap-frontend/src/modules/twap/containers/TwapFormWarnings/warnings/SwapPriceDifferenceWarning.tsx:52]: Nitpick: indentation

```suggestion
          <strong>Maximizing Your Gains! </strong>
```

**Review by alfetopito — APPROVED** [review]
Awesome refactor, thank you!

Some nitpick comments with my manual linting.

**Pull request merged**